models/stock_picking.py

```diff
@@ -280,13 +280,15 @@ def update_picking(
                 An array with the products information to be marked as done,
                 where each dictionary contains: product_barcode, qty and
                 serial numbers if needed
-
+            @param (optional) real_time_update: Boolean
+                Used to specify if the update should be should be processed
+                imidately or on confirmation of the picking.
         """
         Location = self.env['stock.location']
         Package = self.env['stock.quant.package']
-
         self.assert_valid_state()
 
+        picking = self
```

Comment:
To make things clear, I would explain what you're doing with this var (i.e. use the instance as it is or call the real time update method depending on the flag) and define it right before its possible reassignment and usage in https://github.com/unipartdigital/udes_core/pull/14/files#diff-6794fd642495f025859302bb313f0dbfR327.

To be honest, I would probably go with making the logic explicit (based on the flag) and avoid generalizing with `picking`, but not strong opinion on that :)